### PR TITLE
Parse the ELF file locally and push to API

### PR DESCRIPTION
**Pull request by mattface**
Ripped the elf.py out from the backend, to use it on the builder to extract the extra data bits from vmlinux.

### CODE REVIEW

**Comment by mattface** [comment]
Don't merge this until we are sure python-pyelftools is installed on all the builders

**Comment by MiloCasagrande** [comment]
Thanks Matt! Looks good!
Before merging this though I also need to update the backend so we have the support there as well.

**Comment by mattface** [comment]
This is ready to be merged, but I'd like to wait on Kevin as it's mostly his build script

**Comment by MiloCasagrande** [comment]
@khilman when you are online, can you take a look if everything is fine and merge?

**Comment by khilman** [comment]
First observation, ansible need an update for the builders to have the python elftools 👍 

```
[...]
found vmlinux, parsing sizes
Traceback (most recent call last):
  File "/home/khilman/bin/build.py", line 416, in <module>
    import elf
  File "/work/kernel/ci/kernelci-build/elf.py", line 16, in <module>
    import elftools.elf.constants as elfconst
ImportError: No module named elftools.elf.constants
```

File: build.py
Comment (khilman) [inline build.py]: Why remove the vmlinux file locally.  

This will cause make to rebuild on subsequent rebuilds, even if source hasn't changed.

Comment (khilman) [reply build.py]: Maybe better is to just remove the shutil.copy() command a bit later that copies vmlinux to the install dir.

**Comment by MiloCasagrande** [comment]
@khilman where did you run that script?
We already updated all the builders to install pyelf...

**Comment by khilman** [comment]
@MiloCasagrande I ran it locally (and also didn't notice that the ansible requirement was already mentioned in the logs, sorry for the noise.

**Comment by MiloCasagrande** [comment]
No worries, I though we missed to update one builder.

**Review by khilman — CHANGES_REQUESTED** [review]

**Comment by mattface** [comment]
Updated. No longer delete the vmlinux, just dont copy it into the install_dir, and a quick cleanup of the http post requests because I found unhandled errors which would result in the build failing.

**Comment by khilman** [comment]
@mattface can you separate out the ELF stuff from the http post stuff into separate commits?

**Comment by mattface** [comment]
If I do it won't be until middle of next week, as I've had enough of kernelci for now :) Not your fault ;)

File: build.py
Comment (khilman) [inline build.py:331]: Shouldn't we also remove the vmlinux_file from the build.json since it's no longer uploaded?

**Review by khilman — CHANGES_REQUESTED** [review]

**Comment by khilman** [comment]
OK, I've cherry-picked this patch and modified it slightly, dropping the post/retry stuff as well as removing the print message and the vmlinux_file field from the build.json.

I've applied it to the master branch, so will close this pull request.  Please submit a follow-up patch for the post/retry stuff.

**Pull request closed without merging**
